pandas/core/generic.py

```diff
@@ -5138,9 +5138,13 @@ def describe_1d(data):
         if self.ndim == 1:
             return describe_1d(self)
         elif (include is None) and (exclude is None):
-            if len(self._get_numeric_data()._info_axis) > 0:
+            ncols_numeric = len(self._get_numeric_data()._info_axis)
+            ncols_bool = len(self._get_bool_data()._info_axis)
+            if ncols_numeric > ncols_bool:
```

Comment:
Thus u don't have to change the line, when ``self.select_dtypes(include=[np.number])`` is empty use ``self``.